nvm.sh

```diff
@@ -217,6 +217,28 @@ nvm_rc_version() {
   nvm_echo "Found '${NVMRC_PATH}' with version <${NVM_RC_VERSION}>"
 }
 
+nvm_find_package_json() {
+  local dir
+  dir="$(nvm_find_up 'package.json')"
+  if [ -e "${dir}/package.json" ]; then
+    nvm_echo "${dir}/package.json"
+  fi
+}
+
+nvm_package_json_semver_expression() {
+  export PKG_JSON_SEMVER_EXPRESSION=''
+  local PKG_JSON_PATH
+  PKG_JSON_PATH="$(nvm_find_package_json)"
+  if [ ! -e "${PKG_JSON_PATH}" ]; then
+    nvm_err "No package.json file found"
+    return 1
+  fi
+  # TODO handle error case
+  # TODO implement without node?
```

Comment:
>  "X installer" to require X to work just doesn't make any sense.

I generally agree, but if we discuss the nuances a bit more we can still find a path.  Not all of nvm would depend on node, just the function that parsed the and this function is not the the `nvm use` function but something akin to `nvm_rc_version()`.  Just as nvm does not crumble on the lack of an optional .nvmrc, nvm would be able to optionally look to package.json with a node dependency.

If I understand the PR and nvm.sh correctly, isn't this is what @edwmurph is doing, limiting the node deps to nvm_package_json_semver_expression()? It could have a regex fall back too and lack of package.json or of node or of a successful regex parsing would be gracefully handled just as the lack of a .nvmrc.

Really, the ability to use the correct version of node from package.json on a system without node seems like a quite obscure edge case.